### PR TITLE
ECER-1116: Work Experience References (FE)

### PR DESCRIPTION
**Pull request by pcatkins**
## Description

- Alert Component
- Work Experience References step in wizard
- Connect FE to BE

## Related Jira Issue(s)

- [ECER-934](https://eccbc.atlassian.net/browse/ECER-934)

## Checklist
- [X] I have tested these changes locally.
- [ ] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceCharacterReferencePreview.vue [hunk 1]
```diff
@@ -71,17 +72,13 @@ export default defineComponent({
...
-    },
-    referenceFirstName() {
-      return "—";
-    },
-    referenceEmail() {
-      return "—";
-    },
-    referencePhoneNumber() {
-      return "—";
+    characterReference(): Components.Schemas.CharacterReference {
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceCharacterReferencePreview.vue:75]: @SoLetsDev **NOTE** connecting Character Reference data to preview step

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+          :rules="[Rules.phoneNumber()]"
+          label="Reference Phone Number (Optional)"
+          variant="outlined"
+          color="primary"
+          maxlength="20"
+          class="my-8"
+        ></v-text-field>
+        <v-text-field
+          v-model="hours"
+          label="Work Experience Hours"
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue:44]: I think we are missing :rules="[Rules.required()]" here.

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+        // Change mode to list
+        this.mode = "list";
+      } else {
+        this.alertStore.setFailureAlert("Please fill out all required fields");
+      }
+    },
+    handleCancel() {
+      // Change mode to list
+      this.mode = "list";
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue:246]: Do we need to also resetFormData here? 

I suspect if we try to add/edit a user. Cancel it and then try to add a new user. Will it retain the info we input previously?

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+
+      this.alertStore.setWarningAlert("You have Deleted your Reference.");
+    },
+    resetFormData() {
+      this.id = null;
+      this.firstName = "";
+      this.lastName = "";
+      this.email = "";
+      this.phoneNumber = "";
+      this.hours = "";
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue:287]: are we missing previousFullName?

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue (hunk 2 above)
Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue:44]: Good catch!

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue (hunk 3 above)
Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue:246]: No we're OK, we reset before add/edit.

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue (hunk 4 above)
Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/inputs/EceWorkExperienceReferences.vue:287]: Yes, it wasn't right. Good catch!

**Review by SoLetsDev — APPROVED** [review]

**Pull request merged**